cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowStatsSLAHttpHandlerTest.java

```diff
@@ -113,5 +117,108 @@ public void testStatistics() throws Exception {
     response = doGet(request);
     Assert.assertEquals(HttpResponseStatus.BAD_REQUEST.getCode(),
                         response.getStatusLine().getStatusCode());
+    metricStore.deleteAll();
+  }
+
+  @Test
+  public void details() throws Exception {
+    deploy(WorkflowApp.class);
+    String workflowName = "FunWorkflow";
+    String mapreduceName = "ClassicWordCount";
+    metricStore.deleteAll();
+    String sparkName = "SparkWorkflowTest";
+
+    Id.Program workflowProgram =
+      Id.Workflow.from(Id.Namespace.DEFAULT, "WorkflowApp", ProgramType.WORKFLOW, workflowName);
+    Id.Program mapreduceProgram =
+      Id.Program.from(Id.Namespace.DEFAULT, "WorkflowApp", ProgramType.MAPREDUCE, mapreduceName);
+    Id.Program sparkProgram =
+      Id.Program.from(Id.Namespace.DEFAULT, "WorkflowApp", ProgramType.SPARK, sparkName);
+
+    RunId workflowRun1 = setupRuns(workflowProgram, mapreduceProgram, sparkProgram, store);
+    for (int i = 0; i < 12; i++) {
+      setupRuns(workflowProgram, mapreduceProgram, sparkProgram, store);
+    }
+    String request = String.format("%s/namespaces/%s/apps/%s/workflows/%s/runs/%s/statistics?count=%s&interval=%s",
+                                   Constants.Gateway.API_VERSION_3, Id.Namespace.DEFAULT,
+                                   WorkflowApp.class.getSimpleName(), workflowProgram.getId(), workflowRun1.toString(),
+                                   "20", "1s");
+
+    HttpResponse response = doGet(request);
+    Map<String, Map<String, Map<String, Long>>> workflowStatistics =
+      readResponse(response, new TypeToken<Map<String, Map<String, Map<String, Long>>>>() { }.getType());
+
+    // This should only return 3 because the 1st, 6th and 11th record were picked as the interval is 5 seconds
```

Comment:
So I changed the test case after this comment. If you notice, I have mentioned that I have to ask someone how to clear the previously run records. Following up on it, I have already talked to Shankar about it. Will add that assert in my next commit after doing some refactoring.